File: DaggerLifeExample1/app/src/main/java/com/gabortrajtler/daggerlifeexample1/MainActivity.java
package com.gabortrajtler.daggerlifeexample1;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.gabortrajtler.daggerlifeexample1.life.DaggerLifeComponent;
import com.gabortrajtler.daggerlifeexample1.life.Life;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Life life = DaggerLifeComponent.create().getLife();
        life.enjoy();
    }
}

File: FlowPractice/app/src/main/java/com/example/flowpractice/MainActivity.kt
package com.example.flowpractice

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.wrapContentHeight
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.sp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.example.flowpractice.ui.theme.FlowPracticeTheme

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            FlowPracticeTheme() {
                val viewModel = viewModel<MainViewModel>()
                val time = viewModel.countDownFlow.collectAsState(initial = 10)

                // A surface container using the 'background' color from the theme
                Surface(modifier = Modifier.fillMaxSize(), color = MaterialTheme.colorScheme.background) {
                    CentralText(time.value.toString())
                }
            }
        }
    }
}

@Composable
fun CentralText(text: String) {
    Text(
        text = text,
        fontSize = 300.sp,
        textAlign = TextAlign.Center,
        modifier = Modifier.wrapContentHeight() // center vertically
    )
}

@Preview(showBackground = true)
@Composable
fun DefaultPreview() {
    FlowPracticeTheme {
        CentralText("10")
    }
}